src/som/vmobjects/SClass.java

```diff
@@ -89,7 +90,12 @@ public SClass(final SObjectWithClass enclosing,
       final NodeFactory<? extends AbstractSerializationNode> factory) {
     this.enclosingObject = enclosing;
     this.context = null;
-    this.serializationRoot = new SerializerRootNode(null, factory.createNode(this));
+    if (VmSettings.SNAPSHOTS_ENABLED) {
+      CompilerDirectives.transferToInterpreter();
+      this.serializationRoot = new SerializerRootNode(null, factory.createNode(this));
```

Comment:
The `transferToInterpreter` makes me wonder whether the `serializationRoot` shouldn't go into the `instanceClassGroup`.
Also, please avoid the generic `factory` name for the parameter, it's confusing in this context.

Reply:
Moving the serializationRoot shouldn't be too much work. Do you wan't me to move it, or add a comment/TODO and consider it later.

Reply:
Conceptually it seems to belong into `instanceClassGroup`. `SClass` should really just provide identity. Everything else should be in the more stable structures. We don't want a separate serializer per instance of `SClass` either, I think.

Reply:
From what i saw ObjectLiterals create a new SClass for each instance. Moving the serializer would be beneficial in that case, and ensure there is only one.

Reply:
Not just object literals, any nested class is potentially instantiated many times.
Inner classes for different outer objects do always have separate identities.